Dart/src/com/jetbrains/lang/dart/ide/formatter/DartFormattingModelBuilder.java

```diff
@@ -19,8 +23,15 @@ public FormattingModel createModel(@NotNull final PsiElement element, @NotNull f
     // element can be DartFile, DartEmbeddedContent, DartExpressionCodeFragment
     final PsiFile psiFile = element.getContainingFile();
     final ASTNode rootNode = psiFile instanceof DartFile ? psiFile.getNode() : element.getNode();
-    final DartBlock rootBlock = new DartBlock(rootNode, null, null, settings);
-    return new DocumentBasedFormattingModel(rootBlock, element.getProject(), settings, psiFile.getFileType(), psiFile);
+    final DartBlock rootBlock;
+    Object wasFormatted = rootNode.getUserData(DartPreFormatProcessor.FORMAT_MARK);
+    if (wasFormatted == DartPreFormatProcessor.FORMAT_MARKER) {
```

Comment:
Yes, but they have not been included in this experimental code. Some unit tests also cause that condition.